### PR TITLE
add integration test mode for factory debug

### PR DESCRIPTION
**Pull request by Benhalor**

### CODE REVIEW

File: src/software/firmware/srcs/battery.cpp [hunk 1]
```diff
@@ -86,3 +86,5 @@ void batteryLoop(uint32_t p_cycleNumber) {
     updateBatterySample();
     updateBatteryState(p_cycleNumber);
 }
+
+uint32_t getBatteryLevel() { return rawBatteryMeanVoltage * 0.0296484375; }
```
Comment (jabby) [inline src/software/firmware/srcs/battery.cpp:90]:
```suggestion
uint32_t getBatteryLevel() { return rawBatteryMeanVoltage * 296484375 / 10000000000; }
```

File: src/software/firmware/srcs/integration_test.cpp [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+#define STEP_BATTERY_TEST 7
+
+#define NUMBER_OF_STATES 8
+
+#define UNGREEDY(is_drawn, statement)                                                              \
+    if (is_drawn == 0) {                                                                           \
+        statement;                                                                                 \
+        is_drawn = 1;                                                                              \
+    }
+
```
Comment (jabby) [inline src/software/firmware/srcs/integration_test.cpp:45]: Should we merge this macro between `integration_test.cpp` and `qualification.cpp`?

File: src/software/firmware/srcs/integration_test.cpp [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    screen.print(line2);
+}
+
+//! This function displays only a message on line
+void displayLine(char msg[], uint8_t line) {
+    screen.setCursor(0, line);
+    screen.print("                    ");
+    screen.setCursor(0, line);
+    screen.print(msg);
+}
```
Comment (jabby) [inline src/software/firmware/srcs/integration_test.cpp:78]: Should we merge this code between `integration_test.cpp` and `qualification.cpp`?

File: src/software/firmware/srcs/battery.cpp (hunk 1 above)
Comment (dsferruzza) [reply src/software/firmware/srcs/battery.cpp:90]: This might overflow

**Pull request merged**